modules/actor/actor-wfrp4e.js

```diff
@@ -1283,125 +1283,155 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
+   *
+   * @return {Promise<void>}
+   */
   async activateDisease(disease) {
     disease.system.duration.active = true;
     disease.system.incubation.value = 0;
-    let msg = game.i18n.format("CHAT.DiseaseIncubation", { disease: disease.name })
+    let msg = game.i18n.format("CHAT.DiseaseIncubation", { disease: disease.name });
+
     try {
-      let durationRoll = (await new Roll(disease.system.duration.value).roll()).total
-      msg += game.i18n.format("CHAT.DiseaseDuration", { duration: durationRoll, unit: disease.system.duration.unit })
+      let durationRoll = (await new Roll(disease.system.duration.value).roll()).total;
+      msg += game.i18n.format("CHAT.DiseaseDuration", { duration: durationRoll, unit: disease.system.duration.unit });
       disease.system.duration.value = durationRoll;
-    }
-    catch (e) {
-      msg += game.i18n.localize("CHAT.DiseaseDurationError")
+    } catch (e) {
+      msg += game.i18n.localize("CHAT.DiseaseDurationError");
     }
 
-    let chatData = game.wfrp4e.utility.chatDataSetup(msg, "gmroll", false)
-    chatData.speaker = { alias: this.name }
-    ChatMessage.create(chatData)
+    let chatData = game.wfrp4e.utility.chatDataSetup(msg, "gmroll", false);
+    chatData.speaker = { alias: this.name };
+    ChatMessage.create(chatData);
   }
 
+  /**
+   * Finishes disease and handles Lingering symptoms.
+   *
+   * @param {ItemWfrp4e} disease
+   *
+   * @return {Promise<void>}
+   */
   async finishDisease(disease) {
-
     let msg = game.i18n.format("CHAT.DiseaseFinish", { disease: disease.name })
 
-    if (disease.system.symptoms.includes("lingering")) {
+    if (disease.system.symptoms.value.includes("lingering")) {
       let lingering = disease.effects.find(e => e.name.includes("Lingering"))
       if (lingering) {
-        let difficulty = lingering.name.substring(lingering.name.indexOf("(") + 1, lingering.name.indexOf(")")).toLowerCase()
-
-        this.setupSkill(game.i18n.localize("NAME.Endurance"), { difficulty }).then(setupData => this.basicTest(setupData).then(async test => {
-          if (test.result.outcome == "failure") {
-            let negSL = Math.abs(test.result.SL)
-            if (negSL <= 1) {
-              let roll = (await new Roll("1d10").roll()).total
-              msg += game.i18n.format("CHAT.LingeringExtended", { duration: roll })
-            }
-            else if (negSL <= 5) {
-              msg += game.i18n.localize("CHAT.LingeringFestering")
-              fromUuid("Compendium.wfrp4e-core.diseases.kKccDTGzWzSXCBOb").then(disease => {
-                this.createEmbeddedDocuments("Item", [disease.toObject()])
-              })
-            }
-            else if (negSL >= 6) {
-              msg += game.i18n.localize("CHAT.LingeringRot")
-              fromUuid("Compendium.wfrp4e-core.diseases.M8XyRs9DN12XsFTQ").then(disease => {
-                this.createEmbeddedDocuments("Item", [disease.toObject()])
-              })
-            }
```

Comment:
In this code block I did two things.

1. I replaced the `.then()` syntax with the `await` syntax, since we are in `async` function and it improves readability of the code
2. I removed the call to deprecated `actor.basicTest(setupData)` in favor of `test.roll()`

**Probably worth mentioning that both UUIDs provided here return `null`, so it needs fixing.**